react/components/RenderProvider.tsx

```diff
@@ -654,6 +654,19 @@ export class RenderProvider extends Component<
     }
   }
 
+  private mergeRouteParams<T extends { params: any }>(
+    matchingPage: T,
+    transientRoute: { params: any }
+  ): T {
+    return {
+      ...matchingPage,
+      params: {
+        ...transientRoute.params,
```

Comment:
If you can use spread, transientRoute is an object, probably T should extend something like that

```ts
T extends { params: Record<string | number | symbol, any> }
```

Reply:
I thought about it but this is how params it's typed on the rest of the code, so 🤷